File: src/main/java/ast/ASTDoWhileStmnt.java
package ast;

public class ASTDoWhileStmnt extends ASTNode {
    ASTExpr cond;
    ASTNodeArray<ASTNode> code;

    public ASTDoWhileStmnt (ASTFileLocation loc, ASTExpr cond, ASTNodeArray<ASTNode> code){
        super(loc);
        this.cond = cond;
        this.code = code;
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTDoWhileStmnt: (cond, code)");
        cond.print(i+1);
        code.print(i+1);
    }
}

File: src/main/kotlin/ast/ASTDoWhileStmnt.kt
package ast

class ASTDoWhileStmnt(loc: ASTFileLocation, val cond: ASTExpr, val code: ASTNodeArray<ASTNode>) : ASTNode(loc), ASTBlockStmnt {

    override fun print(i: Int) {
        printIndent(i)
        println("ASTDoWhileStmnt: (cond, code)")
        cond.print(i + 1)
        code.print(i + 1)
    }

    override fun getBlock(): ASTNodeArray<ASTNode> {
        return code
    }
}